Styles and themes

Nadezhda-Gurova opened the pull request.

Nadezhda-Gurova committed:
Проект Playlist_Maker

Nadezhda-Gurova committed:
Рабочая версия 1

Nadezhda-Gurova committed:
Исправила ошибки

Nadezhda-Gurova committed:
- исправила цвета
- установила марджины на экране настроек

Nadezhda-Gurova committed:
- создала стили

Nadezhda-Gurova committed:
Реализовала экран настроек полностью:
_ Реализовать кнопку «Поделиться приложением»
- Реализовать кнопку «Написать в поддержку»
- Реализовать кнопку «Пользовательское соглашение»

Nadezhda-Gurova committed:
Реализовала ночную тему на главном экране и экране настроек

Nadezhda-Gurova committed:
начала делать SearchActivity

Nadezhda-Gurova committed:
частично сделала SearchActivity

Nadezhda-Gurova committed:
сделала SearchActivity (БЕЗ КРЕСТИКА)

Nadezhda-Gurova committed:
сделала крестик SearchActivity

Nadezhda-Gurova committed:
дописала код в onSaveInstanceState, onRestoreInstanceState

Nadezhda-Gurova committed:
дописала код в onSaveInstanceState, onRestoreInstanceState

faritowich commented on app/src/main/res/values/styles.xml:
```diff
@@ -0,0 +1,63 @@
+        <item name="android:textSize">@dimen/text_size</item>
+        <item name="layout_constraintStart_toStartOf">parent</item>
+        <item name="android:fontFamily">@font/ys_display_regular</item>
+        <item name="android:layout_marginEnd">@dimen/text_icon_indent_settings</item>
```
Лучше внешние отступы, высоту и ширину, веса не выносить в стиль. Если на экране будет много элементов, будет сложнее ориентироваться с их взаимным расположением, когда отвечающие за это атрибуты разнесены по разным файлам - что-то может быть в лейауте, что-то в стиле. В стилях обычно содержат атрибуты, отвечающие за внешний вид вьюхи - цвета, шрифты, фоны и т.д. Бывают исключения, когда, например, мы точно знаем, что у кнопки всегда должна строго определенная высота и ширина - тогда да, имеет смысл выносить

faritowich commented on app/src/main/res/values/dimen.xml:
```diff
@@ -0,0 +1,16 @@
+    <dimen name="left_indent">16dp</dimen>
+    <dimen name="right_indent">16dp</dimen>
+    <dimen name="text_size">16sp</dimen>
+    <dimen name="padding_inside_container">21dp</dimen>
```
Рекомендуется делать размеры кратными 8 (или 4). Это обеспечивает единообразие дизайна, а также связано с тем, что большинство размеров экрана делятся на 8. Иногда эту рекомендацию приходится нарушать, но лучше по возможности ей следовать.
Также см. https://material.io/design/layout/spacing-methods.html#baseline-grid

faritowich commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -0,0 +1,67 @@
+
+class SearchActivity : AppCompatActivity() {
+    companion object {
+        const val SEARCH_INPUT = "SEARCH_INPUT"
```
Если планируем использовать только в данном классе, стоит сделать private

faritowich commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -0,0 +1,67 @@
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_search)
+
+        val clearButton = findViewById<ImageView>(R.id.clearIcon)
```
Раз уж используется lateinit, лучше все переменные, отвечающие за View, объявлять с lateinit

faritowich commented on app/src/main/java/com/example/playlistmaker/SearchActivity.kt:
```diff
@@ -0,0 +1,67 @@
+                    // empty
+                } else {
+                    // empty
+                }
```
если нет никакой логики, лучше убрать

faritowich commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,114 @@
+            tools:ignore="ContentDescription,MissingConstraints"
+            tools:layout_editor_absoluteX="359dp"
+            tools:layout_editor_absoluteY="20dp" />
+    </androidx.constraintlayout.widget.ConstraintLayout>
```
ConstraintLayout отличный инструмент, но также стоит иметь в виду, что иногда более эффективным решением будет использовать вложенные более простые лейауты - frame и linear. На эту тему можно почитать, например, [эту статью](https://medium.com/@denisgolubev1999/%D1%81%D1%80%D0%B0%D0%B2%D0%BD%D0%B5%D0%BD%D0%B8%D0%B5-constraintlayout-%D1%81-%D0%B4%D1%80%D1%83%D0%B3%D0%B8%D0%BC%D0%B8-viewgroup-%D0%B8-%D0%BA%D0%BE%D0%B3%D0%B4%D0%B0-%D0%BE%D1%82-constraintlayout-%D1%81%D1%82%D0%BE%D0%B8%D1%82-%D0%BE%D1%82%D0%BA%D0%B0%D0%B7%D0%B0%D1%82%D1%8C%D1%81%D1%8F-1878460ed3af)

Для таких очень простых элементов намного целесообразнее использовать FrameLayout и просто прижать элементы в началу и концу при помощи gravity

Nadezhda-Gurova replied:
Большое спасибо за статью)

Nadezhda-Gurova committed:
- Исправила экран поиска, исправила цвета в ночной теме
- При нажатии на кнопку Очистить клавиатура скрывается
- В темном режиме текст-подсказки виден
- Текстовки в классе SettingsActivity вынесла в ресурсы
